mitm.go

```diff
@@ -115,7 +116,27 @@ func (p *Proxy) serveConnect(w http.ResponseWriter, r *http.Request) {
 }
 
 func (p *Proxy) cert(names ...string) (*tls.Certificate, error) {
-	return genCert(p.CA, names)
+	var (
+		cache = map[string]tls.Certificate{} // names (joined with ,) -> cert
```

Comment:
Sure. In principle, caching the generated cert is totally fine. Just be prepared to show some measurements of how much it helps speed and what the tradeoff would be in increased memory use. :) For example, before and after results of a benchmark function would be excellent.